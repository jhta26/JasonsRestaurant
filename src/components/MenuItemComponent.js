import React from 'react';

var itemData = [];

export default function MenuItemComponent({
  item = { name: 'N/A', imagePath: '//via.placeholder.com/300x200', price: '' },
  onAddItem
}) {
  function handleClick(event) {
    event.preventDefault();
    onAddItem(item.id);
  }

  return (
    <div className="col s12 m12 l6">
      <div className="card medium cards MenuItemComponent">
        <div className="card-image">
          <img src={item.imagePath ? item.imagePath : 'N/A'} />
        </div>
        <div className="card-content">
          <h5 className="card-title">
            {item.name ? `${item.name}` : 'N/A'}
          </h5>
          <p>
            {item.price ? `${item.price.toFixed(2)}` : 'N/A'}
          </p>
        </div>
        <div className="card-action">
          <a className ='link' href="#" onClick={handleClick}>
            Add Order
          </a>
        </div>
      </div>
    </div>
  );
}
