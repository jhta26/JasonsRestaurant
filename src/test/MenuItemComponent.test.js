import React from 'react';
import { shallow, mount } from 'enzyme';
import { assert } from 'chai';
import MenuItemComponent from '../components/MenuItemComponent'
import renderer from 'react-test-renderer'



const item = { name: 'Some Item A', price: 9.99, imagePath: '//via.placeholder.com/300x200' }
const onAddItem = jest.fn()


describe('Testing MenuItemComponent',()=>{
	

it('should render the menu item name', ()=>{
	const shallowWrapper= shallow(<MenuItemComponent item={item} onAddItem={onAddItem}/>)
	const itemNameWrapper = shallowWrapper.find('.card-title')
	assert.equal(itemNameWrapper.length,1)
	
})
it('should render the menu price', ()=>{
	const shallowWrapper= shallow(<MenuItemComponent item={item} onAddItem={onAddItem}/>)
	let itemPriceWrapper = shallowWrapper.find('.card-content').children('p')
	expect(itemPriceWrapper.text()).toBe('9.99')
	
})
it('should test add Item',()=>{

const shallowWrapper= shallow(<MenuItemComponent item={item} onAddItem={onAddItem}/>)
mount(<MenuItemComponent item={item} onAddItem={onAddItem}/>).find('.link').simulate('click')
expect(onAddItem).toHaveBeenCalled()


})

it('renders correctly',()=>{
const shallowWrapper= shallow(<MenuItemComponent item={item} onAddItem={onAddItem}/>)
var item = { name: 'Some Item A', price: 9.99, imagePath: '//via.placeholder.com/300x200' }
const image = renderer.create(<img src={item.imagePath || '//via.placeholder.com/300x200'} alt="" />).toJSON()
expect(image).toMatchSnapshot()


})


})