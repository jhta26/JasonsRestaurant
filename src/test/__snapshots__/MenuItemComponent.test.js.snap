// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MenuItemComponent renders correctly 1`] = `
<img
  alt=""
  src="//via.placeholder.com/300x200"
/>
`;
